### PR TITLE
Test cases for shared.py

### PR DESCRIPTION
**Pull request by anand-skss**
* Test cases for shared.py

### CODE REVIEW

File: src/tests/test_shared.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                "st_uid": 1000,
+                "st_gid": 0,
+                "st_size": 1021,
+                "st_atime": 1711587560,
+                "st_mtime": 1709449249,
+                "st_ctime": 1709449603,
+            },
+        )
+        fixSensitiveFilePermissions(filename, False)
+        mock_chmod.asset_called_once()
```
Comment (PeterSurda) [inline src/tests/test_shared.py:130]: this looks like a typo, it should be `assert_called_once`

File: src/tests/test_shared.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    from mock import patch, PropertyMock
+
+
+class TestShared(unittest.TestCase):
+    """Test class for shared.py"""
+
+    @patch("pybitmessage.shared.sqlQuery")
+    def test_isaddress_in_myaddressbook(self, mock_sql_query):
+        """Test if address is in MyAddressbook"""
+        address = "BM-2cUGaEcGz9Zft1SPAo8FJtfzyADTpEgU9U"
```
Comment (PeterSurda) [inline src/tests/test_shared.py:29]: maybe use addresses from samples

File: src/tests/test_shared.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        result = checkSensitiveFilePermissions(filename)
+        self.assertTrue(result)
+
+        # test for freebsd system
+        mock_sys.platform = "freebsd7"
+        mock_os_stat.return_value = os.stat_result(
+            sequence=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+            dict={
+                "st_mode": stat.S_IRUSR,
+                "st_ino": 753,
```
Comment (PeterSurda) [inline src/tests/test_shared.py:93]: use constants

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/tests/test_shared.py (hunk 3 above)
Comment (anand-skss) [reply src/tests/test_shared.py:95]: os.stat returns stat_result type object, so, while mocking return value stat_obj is required.

Comment (PeterSurda) [reply src/tests/test_shared.py:96]: I mean the numbers should be constants

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**